File: 680-ValidPalindromeIi/680-ValidPalindromeIi.py
# Last updated: 7/2/2025, 11:52:16 PM
class Solution:
    def validPalindrome(self, s: str) -> bool:
        return self.help(s, 0, len(s) -1 ,0)
    
    def help(self, s:str, i : int, j : int, c : int) -> bool:
        
        
        if c > 1: return False
        
        if i >= j: return True

        if s[i] == s[j]:
            return self.help(s,i+1,j-1,c)
        
        return self.help(s,i,j-1,c+1) or self.help(s,i+1,j,c+1)

File: 680-ValidPalindromeIi/680-ValidPalindromeIi.cpp
// Last updated: 7/2/2025, 11:52:19 PM
class Solution {
public:
    bool validPalindrome(string s) {
      int n = s.size()-1;
        return help(s, 0, n,0);

    }
    bool help(string s, int l, int r,int c)
    {
      if(c > 1) return false;
      while(l <= r){
        if(s[l] == s[r]){
        l++;
        r--;
      }
      else{
      return help(s,l+1,r,c+1) || help(s, l,r-1,c+1);
      }
      }
      return true;

    }
};